### ISSUE TITLE
Upgrade to Nuxt v3

### ISSUE THREAD

**Issue by ajcwebdev** [stepzen-samples/stepzen-nuxt#13]
# Nuxt Version Three

As of just a couple days ago we can now upgrade to the new (years in the making) version [3.0 of Nuxt](https://nuxtjs.org/announcements/nuxt3-beta/). There's a lot of really interesting features and improvements the Nuxt team has been working on including:

* [Vite](https://vite.nuxtjs.org/) by Vue.js author, Evan You is now included as a replacement to [Webpack](https://webpack.js.org/)
* [Nitro](https://v3.nuxtjs.org/concepts/server-engine/) is a new server engine built with [h3](https://github.com/unjs/h3) which is a high performance http framework
* [Nuxt Bridge](https://v3.nuxtjs.org/getting-started/bridge/) provides the ability to incrementally migrate Nuxt v2 applications to v3.

## Do we need Nuxt Bridge?

[Nuxt Bridge](https://v3.nuxtjs.org/getting-started/bridge/) is a forward-compatibility layer that lets you use Nuxt v3 features in a Nuxt v2 project. Because of the relative simplicity at this time of the sample application, my hunch is that we can simply upgrade to v3 without having to change a single line of code. Only way to truly find out though is to actually implement the upgrade and see if anything breaks.

## Changes to development commands in `package.json`

Despite my optimism for the Nuxt code itself remaining stable, it does appear that the dev commands and `package.json` will end up slightly modified since Nuxt v3 is now using Vite instead of Webpack.

### Nuxt v3 has two main commands
1. `dev` - Start the development server
2. `build` - Build production assets

Nuxt v3 becomes a dev dependency with the new Nitro server, so you will need to add [two commands](https://v3.nuxtjs.org/getting-started/commands/) in your `package.json`:

```json
"scripts": {
  "dev": "nuxi dev",
  "build": "nuxi build",
}
```

**Comment by codiini** [comment]
Hi, can I take this up?

**Comment by ajcwebdev** [comment]
Go for it!

**Comment by codiini** [comment]
@ajcwebdev I think I've successfulyy done the migration, but then I try to run the project and I still get the `Cannot read property 'mountains' of undefined` error

**Comment by codiini** [comment]
@ajcwebdev When i run `stepzen start` I do get redirected to the GraphQL editor but I also get an error 
`Could not load https://stpetebeach.stepzen.io/stpetebeach/hello/__graphql`. I have updated my own fork of the project, you can review here: https://github.com/codiini/stepzen-nuxt
![Web capture_18-10-2021_234529_localhost](https://user-images.githubusercontent.com/57962747/137816550-d52e9b03-a0a5-4a78-87b4-320f3b3bd18b.jpeg)

**Comment by ajcwebdev** [comment]
Interesting, are you sure that you're in the root directory of the project when you run `stepzen start`? The provided URL should be: `https://{username}.stepzen.net/api/stepzen-nuxt-tutorial/__graphql`, which in your case would be `https://stpetebeach.stepzen.net/api/stepzen-nuxt-tutorial/__graphql`. This is specified in the `stepzen.config.json` file.

I checked out your fork and was able to run the example successfully, so the Nuxt 3 upgrade seems to be working fine, we just need to figure out what's going on with your StepZen setup.

**Comment by ajcwebdev** [comment]
That means that you have not included your API key in the `.env` file. If you visit your [account page on StepZen](https://stepzen.com/account) you will see an account name, admin key, and API key. You want to include the API key in your `.env` file like so:

```
STEPZEN_API_KEY=stpetebeach::stepzen.net+1000::xxxx
STEPZEN_ENDPOINT=https://stpetebeach.stepzen.net/api/stepzen-nuxt-tutorial/__graphql
```

**Comment by codiini** [comment]
I had done this already, i find it weird that it doesnt work

**Comment by codiini** [comment]
I've double checked that and I used the API key. I have mailed you my details to help me check again

**Comment by ajcwebdev** [comment]
Sure thing, I see how that can be confusing.

**Comment by ajcwebdev** [comment]
Closed by #14
